hydrolib/core/basemodel.py

```diff
@@ -704,14 +706,16 @@ def __init__(
         If the filepath is provided, it is read from disk.
 
         Args:
-            filepath (Optional[Path], optional): The file path. Defaults to None.
+            filepath (Optional[Union[Path,str]], optional): The file path. Defaults to None.
             resolve_casing (bool, optional): Whether or not to resolve the file name references so that they match the case with what is on disk. Defaults to False.
             recurse (bool, optional): Whether or not to recursively load the model. Defaults to True.
         """
         if not filepath:
             super().__init__(*args, **kwargs)
             return
 
+        filepath = FileModel._change_to_path(filepath)
```

Comment:
I was looking at your PR because I was wondering how you solved it in the end. Because I had to use similar logic for the newly introduced path style parameter in PR #361.
I was wondering what are your thought of the following: https://github.com/Deltares/HYDROLIB-core/blob/0986fe1a92309661507828c89739d2eeb7aeb695/hydrolib/core/basemodel.py#L783.

Maybe we can add the file path validation to the UserInputValidation class. 

Moreover, I noticed that in the `_change_to_path` function below the `filepath` is checked for None, but the check above the function call (line 713) already handles this situation.

Reply:
discussed

Reply:
@MRVermeulenDeltares : can you briefly summarize what you and Prisca decided?

Reply:
Discussed what was implemented for #365 and why the none checks were added to the _change_to_path method.
No further impact on issue #365 related to the UserInputValidation.
Discussion can be resolved.